### PR TITLE
PrismaCloudCompute add ComplianceTable script

### PR DESCRIPTION
**Pull request by DinaMeylakh**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-9420)

## Description
Adding the ComplianceTable script to be used within a designated playbook.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/PrismaCloudCompute/ReleaseNotes/1_6_0.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+#### Scripts
+
+##### New: PrismaCloudComputeComplianceTable
+
+- New: Iterate over EnrichedComplianceIssue information in the context data and add the important keys to a table under PrismaCloudCompute.ComplianceTable. (Available from Cortex XSOAR 6.0.0).
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloudCompute/ReleaseNotes/1_6_0.md:6]:
```suggestion
New: Iterate over EnrichedComplianceIssue information in the context data and add the important keys to a table under PrismaCloudCompute.ComplianceTable. (Available from Cortex XSOAR 6.0.0).
```

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+args:
+- description: 'Type of resource to add to the table.'
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml:2]:
```suggestion
- description: Type of resource to add to the table.
```

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  name: resourceType
+  required: true
+  auto: PREDEFINED
+  predefined:
+    - "Host"
+    - "Container"
+    - "Image"
+- description: 'The context path to the enriched compliance issues list.'
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml:10]:
```suggestion
- description: The context path to the enriched compliance issues list.
```

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  name: contextPath
+  required: false
+  type: str
+comment: 'Iterate over EnrichedComplianceIssue information in the context data and add the important keys to a table under PrismaCloudCompute.ComplianceTable.'
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml:14]:
```suggestion
comment: Iterate over EnrichedComplianceIssue information in the context data and add the important keys to a table under PrismaCloudCompute.ComplianceTable.
```

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  type: Dictionary
+- contextPath: PrismaCloudCompute.ComplianceTable.Host.CloudMetadata
+  description: 'Cloud metadata of the host.'
+  type: Dictionary
+- contextPath: PrismaCloudCompute.ComplianceTable.Container.ContainerID
+  description: 'Container id of the container.'
+  type: String
+- contextPath: PrismaCloudCompute.ComplianceTable.Container.ComplianceIssues
+  description: 'Compliance issue records related to the container. Updated in every iteration.'
+  type: Array
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml:37]:
```suggestion
  description: ID of the host.
  type: String
- contextPath: PrismaCloudCompute.ComplianceTable.Host.ComplianceIssues
  description: Compliance issue records related to the host. Updated in every iteration.
  type: Array
- contextPath: PrismaCloudCompute.ComplianceTable.Host.ComplianceDistribution
  description: Compliance distribution of the host.
  type: Dictionary
- contextPath: PrismaCloudCompute.ComplianceTable.Host.CloudMetadata
  description: Cloud metadata of the host.
  type: Dictionary
- contextPath: PrismaCloudCompute.ComplianceTable.Container.ContainerID
  description: ID of the container.
  type: String
- contextPath: PrismaCloudCompute.ComplianceTable.Container.ComplianceIssues
  description: Compliance issue records related to the container. Updated in every iteration.
  type: Array
```

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  description: 'Image id of the image.'
+  type: String
+- contextPath: PrismaCloudCompute.ComplianceTable.Image.ComplianceIssues
+  description: 'Compliance issue records related to the image. Updated in every iteration.'
+  type: Array
+- contextPath: PrismaCloudCompute.ComplianceTable.Image.ComplianceDistribution
+  description: 'Compliance distribution of the image.'
+  type: Dictionary
+- contextPath: PrismaCloudCompute.ComplianceTable.Image.Hosts
+  description: 'Hosts of the image.'
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml:60]:
```suggestion
- contextPath: PrismaCloudCompute.ComplianceTable.Container.ComplianceDistribution
  description: Compliance distribution of the container.
  type: Dictionary
- contextPath: PrismaCloudCompute.ComplianceTable.Container.Hostname
  description: Hostname of the container.
  type: String
- contextPath: PrismaCloudCompute.ComplianceTable.Container.ImageName
  description: Image name of the container.
  type: String
- contextPath: PrismaCloudCompute.ComplianceTable.Container.CloudMetadata
  description: Cloud metadata of the container.
  type: Dictionary
- contextPath: PrismaCloudCompute.ComplianceTable.Image.ImageID
  description: ID of the image.
  type: String
- contextPath: PrismaCloudCompute.ComplianceTable.Image.ComplianceIssues
  description: Compliance issue records related to the image. Updated in every iteration.
  type: Array
- contextPath: PrismaCloudCompute.ComplianceTable.Image.ComplianceDistribution
  description: Compliance distribution of the image.
  type: Dictionary
- contextPath: PrismaCloudCompute.ComplianceTable.Image.Hosts
  description: Hosts of the image.
```

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  type: Array
+- contextPath: PrismaCloudCompute.ComplianceTable.Container.ImageInstances
+  description: 'Image instances of the image.'
+  type: Array
+- contextPath: PrismaCloudCompute.ComplianceTable.Container.CloudMetadata
+  description: 'Cloud metadata of the image.'
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.yml:66]:
```suggestion
  description: Instances of the image.
  type: Array
- contextPath: PrismaCloudCompute.ComplianceTable.Container.CloudMetadata
  description: Cloud metadata of the image.
```

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/README.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+| resourceType | Type of resource to add to the table. |
+| contextPath | The context path to the enriched compliance issues list. |
+
+## Outputs
+
+---
+
+| **Path**                                                            | **Description** | **Type** |
+|---------------------------------------------------------------------| --- | --- |
+| PrismaCloudCompute.ComplianceTable.Host.Hostname                    | Hostname id of the host. | String |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/README.md:28]:
```suggestion
| PrismaCloudCompute.ComplianceTable.Host.Hostname                    | ID of the host. | String |
```

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/README.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+| PrismaCloudCompute.ComplianceTable.Host.ComplianceIssues            | Compliance issue records related to the host. Updated in every iteration. | Array |
+| PrismaCloudCompute.ComplianceTable.Host.ComplianceDistribution      | Compliance distribution of the host. | Dictionary |
+| PrismaCloudCompute.ComplianceTable.Host.CloudMetadata               | Cloud metadata of the host. | Dictionary |
+| PrismaCloudCompute.ComplianceTable.Container.ContainerID            | Container id of the container. | String |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/README.md:32]:
```suggestion
| PrismaCloudCompute.ComplianceTable.Container.ContainerID            | ID of the container. | String |
```

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/README.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+| PrismaCloudCompute.ComplianceTable.Container.ComplianceIssues       | Compliance issue records related to the container. Updated in every iteration. | Array |
+| PrismaCloudCompute.ComplianceTable.Container.ComplianceDistribution | Compliance distribution of the container. | Dictionary |
+| PrismaCloudCompute.ComplianceTable.Container.Hostname               | Hostname of the container. | String |
+| PrismaCloudCompute.ComplianceTable.Container.ImageName              | Image name of the container. | String |
+| PrismaCloudCompute.ComplianceTable.Container.CloudMetadata          | Cloud metadata of the container. | Dictionary |
+| PrismaCloudCompute.ComplianceTable.Image.ImageID                    | Image id of the image. | String |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/README.md:38]:
```suggestion
| PrismaCloudCompute.ComplianceTable.Image.ImageID                    | ID of the image. | String |
```

**Comment by ShirleyDenkberg** [comment]
@ShahafBenYakir Doc review completed.

**Review by ShahafBenYakir — APPROVED** [review]

File: Packs/PrismaCloudCompute/ReleaseNotes/1_6_0.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (DinaMeylakh) [reply Packs/PrismaCloudCompute/ReleaseNotes/1_6_0.md]: https://jira-dc.paloaltonetworks.com/browse/CIAC-9746

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable_test.py [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+@pytest.mark.parametrize('input_context, input_context_in_path, object_type, already_present_output',
+                         [
+                             (HOST_INPUT, HOST_INPUT['PrismaCloudCompute']['ReportHostScan'], 'Host', [HOST_OUTPUT]),
+                             (CONTAINER_INPUT, CONTAINER_INPUT['PrismaCloudCompute']['ContainersScanResults'], 'Container',
+                              [CONTAINER_OUTPUT]),
+                             (IMAGE_INPUT, IMAGE_INPUT['PrismaCloudCompute']['ReportsImagesScan'], 'Image', [IMAGE_OUTPUT])],
+                         ids=['Host', 'Container', 'Image'])
+def test_update_existing_outputs(mocker, input_context, input_context_in_path, object_type, already_present_output):
+    """
+    Given:
```
[the remark this answers is not in the document]
Comment (DinaMeylakh) [reply Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable_test.py:346]: Let me know if its still unclear

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.py [hunk 12]
```diff
@@ -228,24 +239,143 @@ def update_output_obj_with_issues(compliance_obj: ComplianceObject, input_obj_id
...
+    Args:
+        all_new_data (list): List of the new data.
+        output_objs_to_append (dict): Dict of (key: object id) and their (value: new compliance issues).
+        compliance_obj (ComplianceObject): The compliance object the data refers to.
+        grid_id (str): The grid id to update.
+    """
+    demisto.debug(f"Updating grid {grid_id} table with all new:\n{all_new_data}\noutput_objs:\n{output_objs_to_append}")
+    incident = demisto.incident()
+    custom_fields = incident.get("CustomFields", {}) or {}
+    current_table = custom_fields.get(grid_id)
```
Comment (DinaMeylakh) [inline Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable.py:296]: take the current table from main function

File: Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable_test.py [hunk 13]
```diff
@@ -250,7 +422,7 @@ def test_create_and_update_outputs(mocker, input_context, input_context_in_path,
     """
     Given:
         NEW enriched compliance issues with OLD & NEW compliance objects ids in the input context data.
-        Check for each compliance object type (Host, Container, Image).
+        Check for each compliance object type in different cases: Hosts, Containers, Images.
```
Comment (DinaMeylakh) [inline Packs/PrismaCloudCompute/Scripts/PrismaCloudComputeComplianceTable/PrismaCloudComputeComplianceTable_test.py:425]: make this a sentence

**Comment by DinaMeylakh** [comment]
This branch will be merged with https://github.com/demisto/content/pull/32936 and merged to master together.

**Pull request closed without merging**